websites/Y/YTPLR/presence.ts

```diff
@@ -0,0 +1,82 @@
+const presence = new Presence({
+		clientId: "1284161421957136486",
+	}),
+	strings = presence.getStrings({
+		play: "Watching",
+		pause: "Paused",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let videoTitle: string,
+	videoCurrentTime: number,
+	videoDuration: number,
+	videoPaused: boolean,
+	videoPlaylistTotal: number,
+	videoPlaylistID: string,
+	videoPlaylistTitle: string;
+
+interface DataInterface {
+	title: string;
+	currentTime: number;
+	duration: number;
+	paused: boolean;
+}
+
+const enum Assets {
+	Logo = "https://youtube-playlist-randomizer.bitbucket.io/music-icon.png",
+}
+
+presence.on("iFrameData", (data: DataInterface) => {
+	videoTitle = data.title;
+	videoCurrentTime = data.currentTime;
+	videoDuration = data.duration;
+	videoPaused = data.paused;
+	videoPlaylistTotal = parseInt(document.querySelector("#all").textContent);
+	videoPlaylistID = (document.querySelector("#pid") as HTMLInputElement).value;
+	videoPlaylistTitle = document.querySelector("#title").textContent;
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		},
+		[startTimestamp, endTimestamp] = presence.getTimestamps(
+			Math.floor(videoCurrentTime),
+			Math.floor(videoDuration)
+		),
+		[showButtons] = await Promise.all([
+			presence.getSetting<boolean>("buttons"),
+		]);
```

Comment:
As there is only one setting, `Promise.all` is not needed here.
```suggestion
		showButtons = await presence.getSetting<boolean>("buttons");
```